lib/TerrainHandler.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+ * Terrain.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "StdInc.h"
+#include "TerrainHandler.h"
+#include "CModHandler.h"
+#include "CGeneralTextHandler.h"
+
+VCMI_LIB_NAMESPACE_BEGIN
+
+TerrainType * TerrainTypeHandler::loadFromJson( const std::string & scope, const JsonNode & json, const std::string & identifier, size_t index)
+{
+	TerrainType * info = new TerrainType;
+
+	info->id = TerrainId(index);
+
+	if (identifier.find(':') == std::string::npos)
+		info->identifier = scope + ":" + identifier;
+	else
+		info->identifier = identifier;
+
+	info->moveCost = static_cast<int>(json["moveCost"].Integer());
+	info->musicFilename = json["music"].String();
+	info->tilesFilename = json["tiles"].String();
+	info->horseSound = json["horseSound"].String();
+	info->horseSoundPenalty = json["horseSoundPenalty"].String();
+	info->transitionRequired = json["transitionRequired"].Bool();
+	info->terrainViewPatterns = json["terrainViewPatterns"].String();
+
+	VLC->generaltexth->registerString(info->getNameTextID(), json["text"].String());
+
+	const JsonVector & unblockedVec = json["minimapUnblocked"].Vector();
+	info->minimapUnblocked =
+	{
+		ui8(unblockedVec[0].Float()),
+		ui8(unblockedVec[1].Float()),
+		ui8(unblockedVec[2].Float())
+	};
+
+	const JsonVector &blockedVec = json["minimapBlocked"].Vector();
+	info->minimapBlocked =
+	{
+		ui8(blockedVec[0].Float()),
+		ui8(blockedVec[1].Float()),
+		ui8(blockedVec[2].Float())
+	};
+
+	info->passabilityType = 0;
+
+	for(const auto& node : json["type"].Vector())
+	{
+		//Set bits
+		auto s = node.String();
+		if (s == "LAND") info->passabilityType |= TerrainType::PassabilityType::LAND;
```

Comment:
case sensitive comparison?

Reply:
Yes, intended. Typo's or incorrect case should be detected by json validation.